omod/src/main/webapp/fragments/htmlform/enterHtmlForm.gsp

```diff
@@ -65,20 +65,27 @@
 
         <% if (visit) { %>
             <% if (command.context.mode.toString().equals('ENTER') && !visit.isOpen()) { %>
-                // set default date to the visit start date for retrospective visits
-                htmlForm.setEncounterDate(moment('${ ui.dateToISOString(visit.startDate).split('T')[0] }').toDate());
+                <% if (ui.convertTimezones()) { %>
+                    //New encounter for past visit
+                    htmlForm.adjustTimeZoneEncounterDate('${visitStartDatetime}');
+                <% } else {%>
+                    // set default date to the visit start date for retrospective visits
+                    htmlForm.setEncounterDate('${ visitStartDatetime}');
```

Comment:
Shouldn't this be line 69 above as it was before your changes?

Reply:
Not exactly, before we strip the date from time on server side, now we do that on client side, here : 
https://github.com/openmrs/openmrs-module-htmlformentryui/pull/40/files/64a1b28d878640245daa824ad0c0c469f1e5d232#diff-53c10c5591297c3a544b761c5e9856c108393e5d68c715daabf8b4810697b03eR59